public/api.js

```diff
@@ -0,0 +1,14 @@
+const fetchData = (url, cb) => {
+    const xhr = new XMLHttpRequest();
+    xhr.onreadystatechange = function() {
+        if (xhr.readyState === 4) {
+            if (xhr.status === 200) {
+                cb(JSON.parse(xhr.responseText));
+            } else {
+                cb(xhr.status);
+            }
+        }
+    };
+    xhr.open("GET", url);
+    xhr.send();
+}
```

Comment:
Make a new folder called ```explorePage``` for the API request and initialize two script files one for ```fetch.js``` and another for the ```dom.js```, 
then in the fetch file after making the generic function , make a request for your API  in the same file .

Reply:
done